OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignalDbHelper.java

```diff
@@ -204,8 +204,19 @@ private synchronized void internalOnUpgrade(SQLiteDatabase db, int oldVersion) {
       if (oldVersion < 7)
          upgradeToV7(db);
 
-      if (oldVersion < 8)
-         upgradeToV8(db);
+      // Specifically upgrade from DB versions that already have outcome table with session column created
+      // DB v4 is the first version creating outcome table
+      // If we come from earlier DB versions then the outcome table will already have influence columns instead of session
+      // because upgradeToV4 step will use the new outcome table creation query
+      if (oldVersion > 3 && oldVersion < 8)
```

Comment:
Pre the "Fixing the Current Migrations" section of my long overview it might be easier to read if these are early returns in the `upgradeOutcomeTableRevision1To2` etc methods themselves.
https://github.com/OneSignal/OneSignal-Android-SDK/pull/1052#issuecomment-641793749